components/script/dom/bindings/structuredclone.rs

```diff
@@ -5,15 +5,129 @@
 //! This module implements structured cloning, as defined by [HTML]
 //! (https://html.spec.whatwg.org/multipage/#safe-passing-of-structured-data).
 
+use dom::bindings::conversions::root_from_handleobject;
 use dom::bindings::error::{Error, Fallible};
+use dom::bindings::js::Root;
+use dom::bindings::reflector::DomObject;
+use dom::blob::{Blob, BlobImpl};
 use dom::globalscope::GlobalScope;
-use js::jsapi::{HandleValue, MutableHandleValue};
-use js::jsapi::{JSContext, JS_ReadStructuredClone, JS_STRUCTURED_CLONE_VERSION};
-use js::jsapi::{JS_ClearPendingException, JS_WriteStructuredClone};
+use js::jsapi::{Handle, HandleObject, HandleValue, MutableHandleValue, JSAutoCompartment, JSContext};
+use js::jsapi::{JSStructuredCloneCallbacks, JSStructuredCloneReader, JSStructuredCloneWriter};
+use js::jsapi::{JS_ClearPendingException, JSObject, JS_ReadStructuredClone};
+use js::jsapi::{JS_ReadBytes, JS_WriteBytes};
+use js::jsapi::{JS_STRUCTURED_CLONE_VERSION, JS_WriteStructuredClone, JS_WriteUint32Pair};
+use js::jsapi::{MutableHandleObject, TransferableOwnership};
 use libc::size_t;
+use std::os::raw;
 use std::ptr;
 use std::slice;
 
+///TODO: Add Min and Max const to https://github.com/servo/rust-mozjs/blob/master/src/consts.rs?
+///TODO: Determine which value Min and Max should have.
+///NOTE: Current values found at https://dxr.mozilla.org/mozilla-central/source/js/public/StructuredClone.h#323
+#[allow(dead_code)]
+#[repr(u32)]
+enum StructuredCloneTags {
+    Min = 0xFFFF8000,
+    DomBlob = 0xFFFF8001,
+    ///NOTE: To support additional types, add new tags here.
+    Max = 0xFFFFFFFF,
+}
+
+unsafe extern "C" fn read_blob(cx: *mut JSContext,
+                               r: *mut JSStructuredCloneReader,
+                               data: u32) -> *mut JSObject {
```

Comment:
For multiline argument lists, we prefer putting the return type on its own line too:
```rust

unsafe extern "C" fn read_blob(cx: *mut JSContext,
                               r: *mut JSStructuredCloneReader,
                               data: u32)
                               -> *mut JSObject {
```

Same comment a few more times below.